pkg/antctl/raw/multicluster/create/access_token.go

```diff
@@ -59,78 +70,174 @@ func (o *accessTokenOptions) validateAndComplete() error {
-	if err != nil {
+
+	var createErr error
+	createdRes := []map[string]interface{}{}
+	defer func() {
+		if createErr != nil {
+			if err := common.Rollback(cmd, k8sClient, createdRes); err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "Failed to rollback: %v\n", err)
+			}
+		}
+	}()
+
+	serviceAccount := newServiceAccount(memberTokenOpts.serviceAccount, memberTokenOpts.namespace)
+	u, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(serviceAccount)
+
+	createErr = k8sClient.Create(context.TODO(), serviceAccount)
+	if createErr != nil {
+		if !errors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ServiceAccount \"%s\", error: %s\n", memberTokenOpts.serviceAccount, createErr.Error())
+			return createErr
+		}
+		fmt.Fprintf(cmd.OutOrStderr(), "ServiceAccount \"%s\" already exists\n", memberTokenOpts.serviceAccount)
+		createErr = nil
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ServiceAccount \"%s\" created\n", serviceAccount.Name)
+		createdRes = append(createdRes, u)
+	}
+
+	roleBinding := newRoleBinding(memberTokenOpts.roleBinding, memberTokenOpts.serviceAccount, memberTokenOpts.namespace)
+	u, _ = runtime.DefaultUnstructuredConverter.ToUnstructured(roleBinding)
+
+	createErr = k8sClient.Create(context.TODO(), roleBinding)
+	if createErr != nil {
+		if !errors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStderr(), "RoleBinding \"%s\" already exists\n", memberTokenOpts.roleBinding)
+			createErr = nil
+		}
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to create RoleBingding \"%s\", error: %s\n", memberTokenOpts.roleBinding, createErr.Error())
+		return createErr
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "RoleBinding \"%s\" created\n", roleBinding.Name)
+		createdRes = append(createdRes, u)
+	}
+
+	secretName := args[0]
+	secret := newSecret(secretName, memberTokenOpts.serviceAccount, memberTokenOpts.namespace)
+	createErr = k8sClient.Create(context.TODO(), secret)
+	if createErr != nil {
+		if !errors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStderr(), "Secret \"%s\" already exists\n", secretName)
+		}
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to create Secret \"%s\", start rollback\n", secretName)
+		return createErr
+	}
+	// It will take one or two seconds to wait for the Data.token to be created.
+	if err := waitForSecretReady(k8sClient, secretName, memberTokenOpts.namespace); err != nil {
 		return err
 	}
 
-	serviceAccount := corev1.ServiceAccount{
+	if err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: memberTokenOpts.namespace}, secret); err != nil {
+		return err
+	}
+	s := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      accessTokenOpts.serviceAccount,
-			Namespace: accessTokenOpts.namespace,
+			Name:      secretName,
+			Namespace: "member-cluster-namespace",
```

Comment:
I think we should remove "Namespace". "antctl mc join" should use the user provided Namespace. If users create the Secret using the YAML, they can specify the Namespace with kubectl too.

Reply:
Removed the Namespace.